Gems/ROS2/Code/Source/ContactSensor/ROS2ContactSensorComponent.cpp

```diff
@@ -71,12 +72,18 @@ namespace ROS2
         AZ::Entity* entity = nullptr;
         AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, m_entityId);
         m_entityName = entity->GetName();
-
         auto ros2Node = ROS2Interface::Get()->GetNode();
         AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for Contact sensor");
         const auto publisherConfig = m_sensorConfiguration.m_publishersConfigurations["gazebo_msgs::msg::ContactsState"];
         const auto fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
-        m_contactsPublisher = ros2Node->create_publisher<gazebo_msgs::msg::ContactsState>(fullTopic.data(), publisherConfig.GetQoS());
+        ExecuteROS2Context(
```

Comment:
I see some drawbacks to this construct:
- It requires the developer to know about it, as opposed to doing things as they would normally do in ROS. It's boilerplate code for each (exception-throwing) ROS call, removing users from the "ROS developer experience", which is among the most important selling points for O3DE.
- The developer might not use it (in their sensor, project) and the engine will still crash.
- There is no space to be more specific about exceptions to catch instead of all std::exceptions (are we sure we know that the best way to handle them all is the same?). (https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/exceptions/exceptions.hpp) for rclcpp.
- There is no indication on what to do if it throws (behavior is binary). I imagine that someone could make a ROS call that is not fatal if exception is thrown, including accessing some parameters or creating optional/diagnostic publisher. We might want to just warn in some cases.